# Exploring the Algebra of the 01, 1, 0, 001, 00001000111 cycles

## Original (a,b,c) tuple with Up, Down functions for Shorcut Form:

This was the lattice algebra original identified in the lattice. And it is easiest to see how the period cycle is built in to the positive lattice in this form.

$\tau  = (a, b, c)$  
$V({\tau}) = \frac{2^a - c}{3^b}$  
${\tau}_0  = (0, 0, 0)$  
$f_{up}({\tau})  \mapsto (a+1, b, c)$  
$f_{dn}({\tau})  \mapsto (a+1, b+1, 3c + 2^a)$  

We can see that $f_{up}(f_{dn}({\tau})) \mapsto (a+2, b+1, 3c + 2^a)$

And so we have 
$$
f_{up}(f_{dn}({\tau})) = \frac{2^{a+2} - (3c + 2^a)}{3^{b+1}}
$$
$$
f_{up}(f_{dn}({\tau})) = \frac{4\cdot2^{a} - 3c - 2^a}{3^{b+1}}
$$
$$
f_{up}(f_{dn}({\tau})) = \frac{3\cdot2^{a} - 3c}{3 \cdot 3^{b}}
$$
$$
\frac{2^a - c}{3^b} = V({\tau})
$$



# Mapping $\tau$ tuple to the current:

- $a_{\tau} = a_T$
- $b_{\tau} = -b_T$
- $c_{\tau} = 3^{-b_T}\sum_{(c,d)\in L_T}2^c3^d$
- $f_{up} \mapsto F_1$
- $f_{dn} \mapsto F_0$


## Adding root value to period 2 equations:

$V_{r}({\tau}) = \frac{r2^a - c}{3^b}$  

$$
V_{r}(f_{up}(f_{dn}({\tau}))) = \frac{r2^{a+2} - (3c + 2^a)}{3^{b+1}}
$$
$$
V_{r}(f_{up}(f_{dn}({\tau}))) = \frac{4r\cdot2^{a} - 3c - 2^a}{3^{b+1}}
$$
$$
V_{r}(f_{up}(f_{dn}({\tau}))) = \frac{(4r-1)\cdot2^{a} - 3c}{3 \cdot 3^{b}}
$$
$$
V_{r}(f_{up}(f_{dn}({\tau}))) = \frac{(r-1)\cdot2^{a}}{3^{b+1}} + V_{r}({\tau})
$$

And we see that unless $r=1$, we will not get a cycle with period 2.

# $V_{-1}$, "0" cycle

$V_{-1}({\tau}) = \frac{-2^a - c}{3^b}$  

$f_{dn}({\tau})  \mapsto (a+1, b+1, 3c - 2^a)$ 

$$
V_{-1}(f_{dn}({\tau})) = \frac{-2^{a+1} - 3c - 2^a}{3^{b+1}}
$$
$$
V_{-1}(f_{dn}({\tau})) = \frac{-2 \cdot 2^{a} - 3c - 2^a}{3 \cdot 3^{b}}
$$
$$
V_{-1}(f_{dn}({\tau})) = \frac{-3 \cdot 2^{a} - 3c}{3 \cdot 3^{b}}
$$
$$
V_{-1}(f_{dn}({\tau})) = \frac{-2^{a} - c}{3^{b}}
$$
$$
V_{-1}(f_{dn}({\tau})) = V_{-1}({\tau})
$$


## Adding root value to $V_{-1}$, "0" cycle equations:

$$
V_{r}(f_{dn}({\tau})) = \frac{r \cdot 2^{a+1} - 3c - 2^a}{3^{b+1}}
$$
$$
V_{r}(f_{dn}({\tau})) = \frac{r \cdot 2 \cdot 2^{a} - 3c - 2^a}{3 \cdot 3^{b}}
$$
$$
V_{r}(f_{dn}({\tau})) = \frac{((r+1) -3\cdot 2)2^{a} - 3c}{3^{b+1}}
$$
$$
V_{r}(f_{dn}({\tau})) = \frac{r+1}{3^{b+1}} +  V_{-1}({\tau})
$$

And we see we can only have a cycle when $r = -1$

# $V_{-5}$, "001" cycle

$f_{up}(f_{dn}(f_{dn}({\tau}))) \mapsto (a+3, b+2, 3(3c + 2^a) + 2^{a+1})$  
$\mapsto (a+3, b+2, 9c + 5 \cdot 2^a )$


$V_{-5}({\tau}) = \frac{-5 \cdot 2^a - c}{3^b}$  

$$
V_{-5}(f_{up}(f_{dn}(f_{dn}({\tau})))) = \frac{-5 \cdot 2^{a+3} - (9c + 5 \cdot 2^{a})}{3^{b+2}}
$$
$$
V_{-5}(f_{up}(f_{dn}(f_{dn}({\tau})))) = \frac{-9\cdot 5 \cdot 2^{a} - 9c }{9 \cdot 3^{b}}
$$
$$
\frac{-5 \cdot 2^a - c}{3^b} = V_{-5}({\tau})
$$  




# $V_{-17} $ "00001000111" cycle

In [30]:
accum = 0
for i in [0, 1, 2, 3, 5, 6, 7]:
    accum = 3*accum + (2**i)
    print(accum)


1
5
19
65
227
745
2363


In [31]:
2363 / 17

139.0


- 0: $(a+1, b+1,~ 3c + 2^a)$
- 00: $(a+2, b+2,~ 9c + 5 \cdot 2^a)$
- 000: $(a+3, b+3,~ 27c + 19 \cdot 2^a)$
- 0000: $(a+4, b+4,~ 81c + 65 \cdot 2^a)$
- 00001: $(a+5, b+4,~ 81c + 65 \cdot 2^a)$
- 000010: $(a+6, b+5,~ 243c + 227 \cdot 2^a)$
- 0000100: $(a+7, b+6,~ 729c + 745 \cdot 2^a)$
- 00001000: $(a+8, b+7,~ 2187c + 2363 \cdot 2^a)$
- 000010001: $(a+9, b+7,~ 2187c + 2363 \cdot 2^a)$
- 0000100011: $(a+10, b+7,~ 2187c + 2363 \cdot 2^a)$
- 00001000111: $(a+11, b+7,~ 2187c + 2363 \cdot 2^a)$

$(a+11, b+7,~ 3^7 c + (17*139) \cdot 2^a)$

$$
V_{-17}(f_{00001000111}(\tau)) = \frac{-17 \cdot 2^{a+11} - 3^7 c - 17 \cdot 139 \cdot 2^a}{3^{b+7}}
$$
$$
V_{-17}(f_{00001000111}(\tau)) = \frac{-17 \cdot (2^{11} + 139)2^a - 3^7 c}{3^{b+7}}
$$
$$
V_{-17}(f_{00001000111}(\tau)) = \frac{-17 \cdot (2187)2^a - 3^7 c}{3^{b+7}}
$$
$$
V_{-17}(f_{00001000111}(\tau)) = \frac{-17 \cdot (3^7)2^a - 3^7 c}{3^{b+7}}
$$
$$
V_{-17}(f_{00001000111}(\tau)) = \frac{-17 \cdot 2^a - c}{3^{b}} = V_{-17}(\tau)
$$


# For any given suffix, we should be able to compute the required root ...

In [41]:
1/((2**2) - 3 )

1.0

In [38]:
5/((2**3) - 9 )

-5.0

In [39]:
2363/((2**11) - 2187)

-17.0

# $V_r = (1+r)V_0 - r \cdot V_{-1}$.

Which leads to:

$$
r = \frac{V_0}{1 - (V_0 - V_{-1})}
$$

In [43]:
11*3*2 *2

132

In [44]:
-17*11, -5*3, -1*1, 0, 1*2

(-187, -15, -1, 0, 2)